### DISCUSSION TITLE
Error : UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe8 in position 262: invalid continuation byte

### DISCUSSION THREAD

**Discussion by Gaugo47** [openai/whisper#2241] [Q&A]
Hi, I'm new to this domain but i try to  build an AI like Alexa that work fully locally, I tried to use whisper to transcript my audio to text but when I try to start the program I got this error : "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe8 in position 262: invalid continuation byte" and I don't understand why. I have installed ffmpeg and python-ffmpeg. 
Here is the code : 

#-*- coding: utf-8 -*-
from re import S
import pyaudio
import wave
import tempfile
import os
import sounddevice as sd
import numpy as np
import scipy.io.wavfile as wav
from AI_discussion import SpeechToText

def enregistrer_audio():
    # Paramètres d'enregistrement
    CHUNK = 1024
    FORMAT = pyaudio.paInt16
    CHANNELS = 2
    RATE = 44100
    RECORD_SECONDS = 5
    
    # Définir l'emplacement du répertoire temporaire
    tempdir = tempfile.gettempdir()  # Utiliser le répertoire temporaire par défaut du système
    os.environ['TMPDIR'] = tempdir

    # Créer un fichier temporaire
    temp = tempfile.NamedTemporaryFile(suffix=".wav", dir=tempdir)

    # Initialiser PyAudio
    p = pyaudio.PyAudio()

    try:
        stream = p.open(format=FORMAT,
                        channels=CHANNELS,
                        rate=RATE,
                        input=True,
                        frames_per_buffer=CHUNK)

        print("* Enregistrement...")

        frames = []

        for i in range(0, int(RATE / CHUNK * RECORD_SECONDS)):
            try:
                data = stream.read(CHUNK)
                frames.append(data)
            except IOError as e:
                print(f"Erreur lors de la lecture du flux audio : {e}")

        print("* Fin de l'enregistrement")

        stream.stop_stream()
        stream.close()
    finally:
        p.terminate()

        # Écrire les données dans le fichier WAV
    try:
        with wave.open(temp.name, 'wb') as wf:
            wf.setnchannels(CHANNELS)
            wf.setsampwidth(p.get_sample_size(FORMAT))
            wf.setframerate(RATE)
            wf.writeframes(b''.join(frames))
    except IOError as e:
        print(f"Erreur lors de l'ecriture du fichier WAV : {e}")
        temp.close()
        os.remove(temp.name)
        raise

    print(f"Le fichier WAV temporaire a ete cree a l'emplacement suivant : {temp.name}")
    # Le fichier sera automatiquement supprimé ici, lorsque l'objet 'temp' sera détruit et le fichier fermé
    
    # Retourner le fichier temporaire
    return temp

def jouer_audio(wav_temp):
    try:
        # Lire le fichier WAV
        rate, data = wav.read(wav_temp.name)

        # Jouer l'audio
        sd.play(data, rate)

        # Attendre que l'audio soit fini avant de fermer le fichier
        sd.wait()
    except Exception as e:
        print(f"Erreur lors de la lecture ou de la lecture de l'audio : {e}")
    finally:
        # Fermer le fichier temporaire
        wav_temp.close()
        os.remove(wav_temp.name)

# Utilisation de la fonction
wav_temp = enregistrer_audio()
result = SpeechToText("tempdir") 
result["text"]

and here is the code from Ai_discussion ( I know the are not really accurate...) : 

import whisper

class SpeechToText():
    
    def transcription(audio):
        
        model = whisper.load_model("base")
        result = model.transcribe("audio")
        print(result["text"])

If someone could help me it would be great thanks !
(btw please excuse my english)

**Comment by phineas-pta** [comment]
#302

**Reply by Gaugo47** [reply]
Yes thanks but I saw this and I already tried using wave, but maybe I used it wrong
